src/transformers/models/fastspeech2/tokenization_fastspeech2.py

```diff
@@ -0,0 +1,145 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for FastSpeech2."""
+import json
+import os
+from typing import Optional, Tuple
+
+from ...file_utils import requires_backends
+from ...tokenization_utils import PreTrainedTokenizer
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.json"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "fastspeech2": "https://huggingface.co/fastspeech2/resolve/main/vocab.txt",
+    },
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "fastspeech2": 1024,
+}
+
+
+class FastSpeech2Tokenizer(PreTrainedTokenizer):
+    """
+    Construct a FastSpeech2 tokenizer. Based on byte-level Byte-Pair-Encoding.
+
+    Args:
+        vocab_file (`str`):
+            Path to the vocabulary file.
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    model_input_names = ["input_ids"]
+
+    def __init__(
+        self,
+        vocab_file,
+        unk_token="<unk>",
+        bos_token="<s>",
+        eos_token="</s>",
+        pad_token="<pad>",
+        do_phonemize=True,
+        preserve_punctuation=False,
+        **kwargs
+    ):
+        requires_backends(self, "g2p_en")
+        import g2p_en
+
+        # make attribute to avoid creating g2p object every call
+        self.g2p = g2p_en.G2p()
+        super().__init__(bos_token=bos_token, eos_token=eos_token, unk_token=unk_token, pad_token=pad_token, **kwargs)
+        self.do_phonemize = do_phonemize
+        self.preserve_punctuation = preserve_punctuation
+        with open(vocab_file, encoding="utf-8") as vocab_handle:
+            self.encoder = json.load(vocab_handle)
+        self.decoder = {v: k for k, v in self.encoder.items()}
+
+    @property
+    def vocab_size(self):
+        return len(self.decoder)
+
+    def get_vocab(self):
+        "Returns vocab as a dict"
+        return dict(self.encoder, **self.added_tokens_encoder)
+
+    def phonemize(self, text):
+        if self.preserve_punctuation:
+            return " ".join("|" if p == " " else p for p in self.g2p(text))
+        res = [{",": "sp", ";": "sp"}.get(p, p) for p in self.g2p(text)]
+        return " ".join(p for p in res if p.isalnum())
+
+    def _tokenize(self, text):
+        """Returns a tokenized string."""
+
+        # make sure whitespace is stripped to prevent <unk>
+        text = text.strip()
+
+        # phonemize
+        if self.do_phonemize:
```

Comment:
Can this ever be `False`?

Reply:
I guess it could be if someone is training a model from scratch, but the pretrained checkpoints all require phonemization.